src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/InterOp/D3DImage.cs

```diff
@@ -252,10 +252,7 @@ public bool TryLock(Duration timeout)
         {
             WritePreamble();
 
-            if (timeout == Duration.Automatic)
-            {
-                throw new ArgumentOutOfRangeException("timeout");
-            }
+            ArgumentOutOfRangeException.ThrowIfEqual(timeout, Duration.Automatic);
```

Comment:
I had to roll this back because Duration does not implement IEquatable<Duration>. That would make sense though, and would be trivial to implement